src/service/discovery.js

```diff
@@ -21,9 +21,12 @@ import api from './config';
  * @param {import('axios').AxiosRequestConfig} axiosOptions
  * @return {FetchDiscoveryDataResponse}
  */
-const fetchDiscoveryDataUser = async (query, axiosOptions = {}) => {
+const fetchDiscoveryDataUser = async (query, isAnon, axiosOptions = {}) => {
   try {
-    const response = await api.get(`/discovery/user/?q=${encodeURIComponent(query)}`, axiosOptions);
+    const response = await api.get(
+      `/discovery/user/?q=${query}${isAnon ? '&allow_anon_dm=true' : ''}`,
+      axiosOptions
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The query parameter in the API endpoint is not being encoded which could lead to issues if it contains special characters. It's a good practice to always encode URI components.
```diff
- `/discovery/user/?q=${query}${isAnon ? '&allow_anon_dm=true' : ''}`,
+ `/discovery/user/?q=${encodeURIComponent(query)}${isAnon ? '&allow_anon_dm=true' : ''}`,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
Fixed